### DISCUSSION TITLE
Tasmota DHCP - vendor code (VCI)

### DISCUSSION THREAD

**Discussion by nitinuae** [arendst/Tasmota#16960] [Q&A] [answered]
Hello
I saw a discussion in espressif that VCI option (60/43) was added in the sdk in 2021.
Does Tasmota dhcp client support that.(As I see the sdk used iis based on different fork from espressiff)
Thank you

**Comment by s-hadinger** [comment]
No it's not supported in Tasmota. These features do not seem used in home networks. Can you share why you need this?

Tasmota uses the standard espressif sdk, but we need occasionally to fix or delay some features, hence the fork

**Comment by nitinuae** [comment]
Well - Its one of the MOST IMPORTANT feature request then. 
Long gone are the days where home networks were simpler versions of business networks. Nowadays, most home networks have more than 255 nodes and need to be secure so that attacker donot use them in botnets (you will recall how home routers with outdated sw is used by botnets)
I want to put all tasmota devices in their IOT vlan.  As managing dhcp by mac is not what I am used to, I put all my IOT devices by their vendor id. The IOT vlan is secured by not allowing internet. The attack vector is reduced to a single router.

Users should be encouraged to put their IoTs in separate Vlan and vendor id dhcp is one of the easiest way to do it. I know Tasmota doesnot allow Peap(SDK limitation), but this vendor id should be adopted(as its in the SDK) and users encouraged to use it.

**Comment by Jason2866** [comment]
There is a very low chance for implementing, since many home routers would make trouble.
The (usefull) feature NTP Server anouncement via DHCP was removed from Tasmota since many cheap routers did fail and delivered wrong settings which generated issues in Tasmota.

**Reply by nitinuae** [reply]
I would argue that most of the users who have acumen of atleast flashing tasmota will be using open source Firewall. All opensource firewalls have this implemented (though not easy to use).

Most of the routers in use (home use) have been adopted from open source and dhcp deamon used is not going to throw a fit if it sees VCI- It may choose to ignore.

It is one of the "good practises" and setting aside some of your resources to implment will be highly appreciated (and the team will be thanked for supporting security)

**Reply by techman83** [reply]
That seems like a pretty niche use case, and I'm not sure what security features that would add? Judging by the traffic on the discussion/issues list, the assumption that most would use an open source firewall probably isn't a good one (and there are vendors that sell with Tasmota preloaded on their hardware).

Tasmota is open source, I'm sure if somewhere sufficiently motivated enough to add the feature, it might be considered. Else one could keep a personal fork up to date. That's what I do for a couple of things I don't have the time to implement correctly or really only suit my own use case.

**Reply by nitinuae** [reply]
Its one of the good practsies I was tring to push the dvelopers for. I am not good enough to fork and try to rewrite the book which is already written pretty well.

**Reply by techman83** [reply]
I've never seen this set of options used for this particular use case, though I'm not 100% sure what the desired outcome you're after for VCI. I've normally used it for giving different phone brands to get different options, never for security purposes.

**Reply by nitinuae** [reply]
If there was PEAP, we wont be having this discussion.... As its the best possible solution. Unfortunately, we dont have PEAP. I cannot find a way to have single SSID, and some clients are directed on peap and tasmotas on non peap(I know that I can assign a vlan on peap failure - but then all failuers will go to that vlan)

**Comment by s-hadinger** [comment, marked as answer]
I'm still not sure why this dhcp option is any useful. All my Tasmota devices are on a separate vlan and a separate wlan SSID. My router assigns ip addresses without needing a new dhcp option. 

Moreover I don't see how you would enter the dhcp option before the wlan connection

**Reply by nitinuae** [reply]
I donot use separate SSID - its like advertising to the world - Guys this is my IOT network. Its not elegant.
I work security by this - All mac in the list (Vendor option) are given DHCP by their separate vlan. 
All trusted devices are given a seprate vlan.
All other are directed to generic vlan.

wlan is not secured - it never can be. Vlans are secured. (you can argue that mac spoofing can be done - but that requires physical presence and physical connection to my wlan)

**Reply by s-hadinger** [reply]
I don't see why having a clear separate SSID is not elegant. It's the simplest solution end to end. 

Also statements like "wlan will never be secure and vlans are secure" makes me doubtful at least. Do you have any strong argument about this?

**Reply by s-hadinger** [reply]
Do I understand that your security relies on MAC addresses?

**Reply by s-hadinger** [reply]
One last thing, you do realize that dhcp options are broadcasted in clear on the entire wlan network?

**Reply by nitinuae** [reply]
Ahh - I knew someone would come to - Yes - for generic access. To get into my servers etc I need to authenticate again with keys. But yes - MAC is the first step (and it can be easily spoofed IF the person is conncted to my network physically.

**Reply by techman83** [reply]
It's pretty trivial to force a VLAN, even across WiFi, I'm not sure what security you're gaining here. Personally I keep my IoT VLAN + SSID very separate as I don't trust the client devices to do the right things. Makes it nice and simple, and I don't have to futz around with DHCP settings, mac lists, etc etc. Just plain old DHCP.

**Reply by nitinuae** [reply]
I am not trying to be security expert here- Just requesting if something that is already there in SDK be implemnted.
Yes - The client will broadcast in clear, and server will reply in clear allowing attacker to know my vlans. All this will require the attacker to be driving by and stopping near the house.

Separate ids are not elegant as I have guests arriving who want to stream to my devices. Do I tell them no not this SSID, but connect to the other SSID.

**Reply by techman83** [reply]
Well your request is in the guise of security, not a niche feature. I'm not sure about your guests, but mine don't particularly have a hard time connecting to the 'Guest' SSID, that has a different passphrase and is on it's own vlan separate from my IoT and personal ones. Not only that, my housemate has his own set of SSIDs on top of that. All separate VLANs, and all firewalled from each other.

**Reply by s-hadinger** [reply]
Yes precisely. Create a guest SSID. Since the iot wlan have a different password, there is no chance that your guest will use it or even know what this is. 

Thanks for your request but I'm afraid that it will not be included in the core Tasmota for now. You are welcome to provide a Pull Request if you want to implement it yourself and share to the community.

**Reply by nitinuae** [reply]
> Yes precisely. Create a guest SSID. Since the iot wlan have a different password, there is no chance that your guest will use it or even know what this is.
> 
> Thanks for your request but I'm afraid that it will not be included in the core Tasmota for now. You are welcome to provide a Pull Request if you want to implement it yourself and share to the community.

Is it possible then to look into the code so that Tasmotas can connect to hidden SSIDs?  Tasmotas since last 1 year or so (when you changed SDK ? I am not sure) have refused to connect to hidden SSIDs.

My intention was not to hide this in guise of security - It was a question (which turned as request).
I do use VCI to pull lots of clients into VLAN. (yes - its Mac based security partially)

**Reply by s-hadinger** [reply]
I have never tried hidden SSID. We don't block anything in Tasmota so chances are it's because of the sdk. 

But please don't use hidden SSID. It's a bad feature that makes legitimate users miserable while not bringing any benefit

**Comment by Ronald1817** [comment]
Is there a way to batch change the IP ofr the MQTT server for all the tasmota devices?
Why do i need that?, because i want to move Domoticz server to IOT VLAN , it's now in Data vlan and am placing firewall rules to isolate vlan to vlan except for management vlan that can reach every other vlan.
But is there a easy way to acomplish this?

**Reply by s-hadinger** [reply]
Please don't hijack this discussion for a completely different topic. Open a new discussion.

**Reply by barbudor** [reply]
Your question is not really related to that discussion and would have been worth a separated discussion
But quickly, if both MQTT server are up, you can issue a group command using the group topic
If you haven't changed the default configuration, the group topic is `tasmotas` so
publish to `cmnd/tasmotas/backlog`  your configuration string like `MqttHost Xxxxx; MqttUser Xxxxx ... etc`
If you don't have to change user, password but only the host, you can do without a backlog as publishing to `cmnd/tasmotas/mqtthost` with payload `Xxxx`

**Reply by s-hadinger** [reply]
Or simply use DNS names for your MQTT gateway, and stop using IP addresses

**Comment by Ronald1817** [comment]
Hi

 

That’s the solution , i have a UBNT Dream machine and created local dns and it woks

In the weekend I am going to visit all the tasmota devise that I have and change the MQTT to dns 

 

Thanx

 

Van: s-hadinger ***@***.***> 
Verzonden: woensdag 24 april 2024 22:19
Aan: arendst/Tasmota ***@***.***>
CC: Ronald1817 ***@***.***>; Comment ***@***.***>
Onderwerp: Re: [arendst/Tasmota] Tasmota DHCP - vendor code (VCI) (Discussion #16960)

 

Or simply use DNS names for your MQTT gateway, and stop using IP addresses

**Comment by Ronald1817** [comment]
Ok thought DHCP options and DNS it fitted but it did not.

 

Van: s-hadinger ***@***.***> 
Verzonden: woensdag 24 april 2024 22:00
Aan: arendst/Tasmota ***@***.***>
CC: Ronald1817 ***@***.***>; Comment ***@***.***>
Onderwerp: Re: [arendst/Tasmota] Tasmota DHCP - vendor code (VCI) (Discussion #16960)

 

Please don't hijack this discussion for a completely different topic. Open a new discussion.
